src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/NhsAckContent.java

```diff
@@ -0,0 +1,29 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@Setter
+public class NhsAckContent {
```

Comment:
Builder pattern would be nice here
```suggestion
@Builder
public class NhsAckContent {
```

Reply:
I've refactored it using a builder, although it looks like some parts of the creation have to be done after building, most notably the message results as they have to be added to the list in the NhsAckContent object. (Although this may just be me not having that much experience using builders). If we think this is fine I'm happy enough to change it

`

        nhsAckContentBuilder.messageStatus(new ArrayList<>());

        NhsAckContent nhsAckContent = nhsAckContentBuilder.build();

        int acceptedMessages = 0;
        int rejectedMessages = 0;
        int totalMessages = messageProcessingResults.size();

        for (MessageProcessingResult messageProcessingResult : messageProcessingResults) {
            NhsAckMessageContent nhsAckMessageContent;
            if (messageProcessingResult instanceof MessageProcessingResult.Success) {
                Success currentMessageResult = (Success) messageProcessingResult;
                MessageHeader messageHeader = currentMessageResult.getMessage().getMessageHeader();
                nhsAckMessageContent = NhsAckMessageContent.builder()
                        .messageNumber(messageHeader.getSequenceNumber())
                        .messageStatus("MA")
                        .build();
                acceptedMessages++;`

Reply:
Yeah, the builder pattern is just a way to construct objects - they don't _have_ to be immutable afterwards (though that is preferred).

Reply:
Done now, for some reason GitHub isn't showing this as outdated.